pkg/sources/adapter/azureservicebussource/adapter.go

```diff
@@ -230,22 +243,56 @@ func connectionStringFromEnvironment(namespace, entityPath string) string {
 //  Both (DataAction):
 //  - Microsoft.ServiceBus/namespaces/messages/receive/action
 func (a *adapter) Start(ctx context.Context) error {
+	maxMessages := 100
 	logging.FromContext(ctx).Info("Listening for messages")
 
-	handle := a.msgRcvr.Listen(ctx, servicebus.HandlerFunc(a.handleMessage))
-	<-handle.Done()
-	return handle.Err()
+loop:
+	for {
+		select {
+		case <-ctx.Done():
+			break loop
+		default:
+			messages, err := a.msgRcvr.ReceiveMessages(ctx, maxMessages, nil)
+			if err != nil {
+				return fmt.Errorf("error receiving messages: %w", err)
+			}
+			for _, m := range messages {
+				body, err := m.Body()
+				if err != nil {
+					return fmt.Errorf("error getting message data: %w", err)
+				}
+
+				msg := &Message{
+					Data:            body,
+					ReceivedMessage: *m,
+					LockToken:       stringifyLockToken((*uuid.UUID)(&m.LockToken)),
+				}
+
+				err = a.handleMessage(ctx, msg)
+				if err != nil {
+					return fmt.Errorf("error processing message: %w", err)
+				}
+
+				err = a.msgRcvr.CompleteMessage(ctx, m)
+				if err != nil {
+					return fmt.Errorf("error completing message: %w", err)
+				}
+			}
+		}
+	}
+
+	return nil
 }
 
 // handleMessage satisfies servicebus.HandlerFunc.
```

Comment:
This comment is outdated. Suggestion:

```suggestion
// handleMessage handles a single Service Bus message.
```